### DISCUSSION TITLE
RLS & Stripe Customer ID

### DISCUSSION THREAD

**Discussion by sdbxsdb** [supabase/supabase#9095] [Questions]
In short when RLS is DISABLED I can add a new user and give them their Stripe customerID no problem but when RLS is ENABLED I can't.  

Would I be right in thinking I need to write a new policy to allow the create-stripe-customer API to be able to write to the DB?

SQL is in no way my strong point so any dumbed-down answers would be appreciated!!

Thanks!

**Comment by sdbxsdb** [comment]
I'm now thinking this isn't possible since the request is on the server so maybe I need to bypass RLS to allow this customer to get their ID from stripe....

**Reply by mansueli** [reply]
Have you tried to use a [RPC](https://supabase.com/docs/reference/javascript/rpc) function to achieve this?

Another alternative would be deploying an [Edge Function](https://supabase.com/docs/guides/functions) for it. Both should work, you can even use your server to make the write request using the **service_role** key as it bypasses the RLS.

**Reply by sdbxsdb** [reply]
I've tried using the service_role to no avail but I'm not sure if I'm using it correctly.

I have also got a function for handling a new user but I'm not familiar with SQL.  I've attached a screen shot of the new user [function[](url)](url)
![Screenshot 2022-09-22 at 15 00 45](https://user-images.githubusercontent.com/69519326/191767969-9f491db1-c13e-4625-b5ef-a66014f0e533.png)

**Comment by sdbxsdb** [comment]
I've just realised my req.body is empty so there is no email associated with the new strip customer..  I think this is because I'm using passwordless magic link log in.  Would that make sense?
